[Zenkoh1] iP

Zenkoh1 opened the pull request:
# BobbyBot

BobbyBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

Features:

- [ ]  Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Zenkoh1 committed:
Change BotName to BobbyBot and change output

Zenkoh1 committed:
Allow bot to reply to user

Zenkoh1 committed:
Add ability to store and display stored text entered by user

Zenkoh1 committed:
Add Task class

Zenkoh1 committed:
Add ability to mark and unmark tasks as done

Zenkoh1 committed:
Add event/deadline/todo task types

Zenkoh1 committed:
Add input and expected output for testing

Zenkoh1 committed:
Add input validation and exceptions

Zenkoh1 committed:
Add delete functionality

Zenkoh1 committed:
Fix switch case indentation

Zenkoh1 committed:
Add TaskFile reader/writer

Zenkoh1 committed:
Merge branch 'branch-Level-7'

Zenkoh1 committed:
Fix outputting of tasks into data file

Zenkoh1 committed:
Merge branch 'master' into branch-Level-7

Zenkoh1 committed:
Add date parsing for the tasks

Zenkoh1 committed:
Package code and use more OOP

Zenkoh1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Zenkoh1 committed:
Add JUnit test cases

Zenkoh1 committed:
Configure build.gradle for bobbybot

Zenkoh1 committed:
Add javaDoc comments to most of the public methods

Zenkoh1 committed:
Fix some names and imports to adhere to coding standard

Zenkoh1 committed:
Add new find command

Zenkoh1 committed:
Merge branch 'branch-A-JavaDoc'

Zenkoh1 committed:
Merge coding standard into master

Zenkoh1 committed:
Merge find command into master

Zenkoh1 committed:
Add JavaDoc Comments for Command Classes

Zenkoh1 committed:
Add checkstyle configs

Zenkoh1 committed:
Merge branch 'branch-A-CheckStyle'

Zenkoh1 committed:
Fix all coding style

Zenkoh1 committed:
Add JavaFX UI

Zenkoh1 committed:
Merge branch 'branch-Level-10'

Zenkoh1 committed:
Change MainWindow to use the fx:root construct

kngys commented on src/main/java/bobbybot/Parser.java:
```diff
@@ -0,0 +1,53 @@
+        String argument = inputArray.length > 1 ? inputArray[1].trim() : "";
+        switch (command) {
+        case "bye":
+            return new CommandBye(argument);
```
Should there be a break statement here?

kngys commented on src/main/java/bobbybot/TaskList.java:
```diff
@@ -0,0 +1,114 @@
+package bobbybot;
+import java.util.ArrayList;
```
Consider adding a line before this import statement?

kngys commented on src/main/java/bobbybot/TaskList.java:
```diff
@@ -0,0 +1,114 @@
+     */
+    public TaskList findTasks(String query) {
+        ArrayList<Task> matchingTasks = new ArrayList<>();
+        for (Task task : tasks) {
+            if (task.getDescription().contains(query)) {
+                matchingTasks.add(task);
+            }
+        }
```
Consider using the following form?
```java
for (initialization; condition; update) {
    statements;
}

```

kngys reviewed (commented):
Just some minor suggestions about coding standards, overall looks good to me!

Zenkoh1 committed:
Add assertions to the code

There are currently no assertions in the code.

Assertions will help to detect potential bugs as early
as possible, to prevent bugs from appearing in production code.

Let's add some Java assert statements to ensure that the
assumptions we make about the code is actually true.

Zenkoh1 committed:
Refactor to adhere to coding quality standards

Zenkoh1 committed:
Merge pull request #3 from Zenkoh1/branch-A-Assertions

There are currently no assertions in the code.

Assertions will help to detect potential bugs as early as possible, to prevent bugs from appearing in production code.

Let's add some Java assert statements to ensure that the assumptions we make about the code is actually true.

Zenkoh1 committed:
Merge pull request #4 from Zenkoh1/branch-A-CodeQuality

The quality of the code currently is not up to standard.

Good coding quality will help future me and other programmers
understand the code better. 

Let's refactor some code to adhere to the coding standards.

Zenkoh1 committed:
Add undo functionality

Zenkoh1 committed:
Merge branch 'branch-C-Undo'

sheenkerr commented on src/main/java/bobbybot/BobbyBot.java:
```diff
@@ -0,0 +1,68 @@
+    public BobbyBot() {
+        ui = new Ui();
+        storage = new Storage(FILE_PATH);
+        try {
```
I think it would be better for you to extract the try catch statement into another function for better readability

sheenkerr commented on src/main/java/bobbybot/tasks/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+    public static final String TASK_TYPE = "D";
+
+    private static final DateTimeFormatter DEADLINE_DATE_FORMAT = DateTimeFormatter.ofPattern("MMM d yyyy");
+    private final LocalDate by;
```
Maybe instead of `by `you can use the variable name `completionDate`

sheenkerr reviewed (commented):
Overall your code looks very readable and clean. I found your use of abstractions very smart and would not have thought of abstracting the different commands in their own class. 👍💯

Zenkoh1 committed:
Rename some Memento fields and methods

Zenkoh1 committed:
Update JUnit tests to align with code changes

Zenkoh1 committed:
Add User Guide

Zenkoh1 committed:
Fix User Guide Format

Zenkoh1 committed:
Add links for feature list

Zenkoh1 committed:
Resize the ui image

Zenkoh1 committed:
Revert image resizing

Zenkoh1 committed:
Fix command format in user guide

Zenkoh1 committed:
Fix some Javadoc header comments

Zenkoh1 committed:
Fix undo feature
